tests_integration/test_streamlit.py

```diff
@@ -106,6 +106,30 @@ def test_streamlit_deploy(
     assert row_from_snowflake_session(expect) == []
 
 
+@pytest.mark.integration
+def test_streamlit_deploy_with_imports(
+    runner,
+    snowflake_session,
+    test_database,
+    _new_streamlit_role,
+    project_directory,
+    alter_snowflake_yml,
+):
+
+    # This work because uploading the imports artifact because
+    # deploying streamlit does not start the app.
+    with project_directory(f"streamlit_v2"):
+        alter_snowflake_yml(
+            "snowflake.yml",
+            "entities.my_streamlit.imports",
+            ["@stage/foo.py", "@stage/bar.py"],
+        )
+        result = runner.invoke_with_connection_json(
+            ["streamlit", "deploy", "--replace"]
+        )
+        assert result.exit_code == 0
```

Comment:
Could you check if imports are in query?

Reply:
Aded